packages/record/src/recorder.ts

```diff
@@ -1,17 +1,38 @@
+/* eslint-disable jsdoc/no-types */
+/* eslint-disable jsdoc/no-defaults */
 import Data from "@lookit/data";
 import lookitS3 from "@lookit/data/dist/lookitS3";
 import autoBind from "auto-bind";
 import { JsPsych } from "jspsych";
 import { NoStopPromiseError, RecorderInitializeError } from "./error";
+// import MicCheckProcessor from './mic_check';  // TO DO: fix or remove this
+
+/**
+ * A valid CSS height/width value, which can be a number, a string containing a
+ * number with units, or 'auto'.
+ */
+type CSSWidthHeight =
+  | number
+  | `${number}${"px" | "cm" | "mm" | "em" | "%"}`
+  | "auto";
 
 /** Recorder handles the state of recording and data storage. */
 export default class Recorder {
   private blobs: Blob[] = [];
   private localDownload: boolean =
     process.env.LOCAL_DOWNLOAD?.toLowerCase() === "true";
-  private s3?: lookitS3;
+  private s3: lookitS3 | null = null;
   private filename: string;
-  private stopPromise?: Promise<void>;
+  private stopPromise: Promise<void> | null = null;
```

Comment:
I used `null` so that I could set these back to `null` when the Recorder is destroyed (otherwise I think these could just be defined in the constructor and would not need an "or undefined" in their types). But I'll double check to see if it actually makes sense to set these back to `null` in the `destroy` method.

Reply:
Ok I checked this, and I did switch to `undefined` for the `stopPromise`. 
But I think it's useful for `s3` and `processorNode` to be `null` rather than `undefined` so that they can be set back to `null` in the recorder's `destroy` method. 
For S3, I _think_ this will help clear up any video blob data that could still be stored by the S3 instance (completing any in progress uploads should also do this, but I'm not sure if that covers all cases).
For the processor node, it can be stopped by both the `destroy` method and the mic check itself. So setting it back to `null` is helpful to check that it exists before trying to stop it.
I could be wrong about both of these things - let me know!